inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java

```diff
@@ -471,70 +471,68 @@ private DataConfig getDataConfig(StreamSourceEntity entity, int op) {
+                    clusterInfo.setUrl(cluster.getUrl());
+                    clusterInfo.setToken(cluster.getToken());
+                    clusterInfo.setMqType(cluster.getType());
+                    clusterInfo.setParams(JsonUtils.parseObject(cluster.getExtParams(), HashMap.class));
+                    mqSet.add(clusterInfo);
                 }
-
-                String topic = String.format(InlongConstants.PULSAR_TOPIC_FORMAT,
-                        tenant, mqResource, streamEntity.getMqResource());
-                DataProxyTopicInfo topicConfig = new DataProxyTopicInfo();
-                topicConfig.setInlongGroupId(groupId + "/" + streamId);
-                topicConfig.setTopic(topic);
-                dataConfig.setTopicInfo(topicConfig);
-            } else if (MQType.TUBEMQ.equals(mqType)) {
-                DataProxyTopicInfo topicConfig = new DataProxyTopicInfo();
-                topicConfig.setInlongGroupId(groupId);
-                topicConfig.setTopic(mqResource);
-                dataConfig.setTopicInfo(topicConfig);
+                dataConfig.setMqClusters(mqSet);
+
+                // add topic setting
+                String mqResource = groupEntity.getMqResource();
+                String mqType = groupEntity.getMqType();
+                if (MQType.PULSAR.equals(mqType) || MQType.TDMQ_PULSAR.equals(mqType)) {
+                    // first get the tenant from the InlongGroup, and then get it from the PulsarCluster.
+                    InlongPulsarDTO pulsarDTO = InlongPulsarDTO.getFromJson(groupEntity.getExtParams());
+                    String tenant = pulsarDTO.getTenant();
+                    if (StringUtils.isBlank(tenant)) {
+                        // If there are multiple Pulsar clusters, take the first one.
+                        // Note that the tenants in multiple Pulsar clusters must be identical.
+                        PulsarClusterDTO pulsarCluster = PulsarClusterDTO.getFromJson(
+                                mqClusterList.get(0).getExtParams());
+                        tenant = pulsarCluster.getTenant();
+                    }
+
+                    String topic = String.format(InlongConstants.PULSAR_TOPIC_FORMAT,
+                            tenant, mqResource, streamEntity.getMqResource());
+                    DataProxyTopicInfo topicConfig = new DataProxyTopicInfo();
+                    topicConfig.setInlongGroupId(groupId + "/" + streamId);
+                    topicConfig.setTopic(topic);
+                    dataConfig.setTopicInfo(topicConfig);
+                } else if (MQType.TUBEMQ.equals(mqType)) {
+                    DataProxyTopicInfo topicConfig = new DataProxyTopicInfo();
+                    topicConfig.setInlongGroupId(groupId);
+                    topicConfig.setTopic(mqResource);
+                    dataConfig.setTopicInfo(topicConfig);
+                }
+            } else {
+                dataConfig.setSyncSend(0);
```

Comment:
Why set the syncSend=0 when inlong_group or inlong_stream was null?

Reply:
This set to 0 was unnecessary, I remove it.